### PR TITLE
mach: Extract binary select into common_command_arguments

### PR DESCRIPTION
**Pull request by sagudev**
Multiple commands support specifying binary, this will unify them all so we can use any binary on any such commands (I think nightly binaries needs to start using GitHub, but this is for followup).

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes do not require tests because they are mach changes

### CODE REVIEW

File: python/servo/testing_commands.py [hunk 1]
```diff
@@ -322,10 +322,9 @@ def test_wpt_android(self, build_type: BuildType, binary_args=None, **kwargs):
         return self._test_wpt(build_type=build_type, android=True, **kwargs)
 
     @CommandBase.allow_target_configuration
-    def _test_wpt(self, build_type: BuildType, with_asan=False, **kwargs):
+    def _test_wpt(self, build_type: BuildType, binary, **kwargs):
         # TODO(mrobinson): Why do we pass the wrong binary path in when running WPT on Android?
-        binary_path = self.get_binary_path(build_type, asan=with_asan)
-        return_value = wpt.run.run_tests(binary_path, **kwargs)
+        return_value = wpt.run.run_tests(binary, **kwargs)
```
Comment (sagudev) [inline python/servo/testing_commands.py:327]: Not sure if this works for android.

File: python/servo/testing_commands.py [hunk 2]
```diff
@@ -319,13 +319,12 @@ def test_wpt_android(self, build_type: BuildType, binary_args=None, **kwargs):
             binary_args=self.in_android_emulator(build_type) + (binary_args or []),
             binary=sys.executable,
         )
-        return self._test_wpt(build_type=build_type, android=True, **kwargs)
+        return self._test_wpt(sys.executable, android=True, **kwargs)
```
Comment (sagudev) [inline python/servo/testing_commands.py:322]: Not sure about android.

File: python/servo/command_base.py [hunk 3]
```diff
@@ -520,9 +520,9 @@ def build_env(self):
         return env
 
     @staticmethod
-    def common_command_arguments(build_configuration=False, build_type=False):
+    def common_command_arguments(build_configuration=False, build_type=False, binary=False):
```
Comment (mrobinson) [inline python/servo/command_base.py:523]:
```suggestion
    def common_command_arguments(build_configuration=False, build_type=False, binary_selection=False):
```

File: python/servo/testing_commands.py [hunk 4]
```diff
@@ -764,13 +764,13 @@ def cts(self, repo="https://github.com/gpuweb/cts", checkout="main"):
              category='testing')
     @CommandArgument('params', nargs='...',
                      help="Command-line arguments to be passed through to Servo")
-    @CommandBase.common_command_arguments(build_configuration=False, build_type=True)
-    def smoketest(self, build_type: BuildType, params, with_asan=False):
+    @CommandBase.common_command_arguments(binary=True)
+    def smoketest(self, servo_binary, params):
```
Comment (mrobinson) [inline python/servo/testing_commands.py:768]:
```suggestion
    def smoketest(self, servo_binary: str, params):
```

File: python/servo/testing_commands.py [hunk 5]
```diff
@@ -484,11 +484,11 @@ def jquery_test_runner(self, cmd, build_type: BuildType):
             ["git", "-C", jquery_dir, "pull"])
 
         # Check that a release servo build exists
-        bin_path = path.abspath(self.get_binary_path(build_type))
+        bin_path = path.abspath(binary)
 
         return call([run_file, cmd, bin_path, base_dir])
 
-    def dromaeo_test_runner(self, tests, build_type: BuildType):
+    def dromaeo_test_runner(self, tests, binary):
```
Comment (mrobinson) [inline python/servo/testing_commands.py:491]: Please add typing information here.

File: python/servo/testing_commands.py [hunk 6]
```diff
@@ -469,7 +469,7 @@ def compare_dromaeo(self, params):
                     print("{}|{}|{}|{}".format(a1.ljust(width_col1), str(b1).ljust(width_col2),
                           str(c1).ljust(width_col3), str(d1).ljust(width_col4)))
 
-    def jquery_test_runner(self, cmd, build_type: BuildType):
+    def jquery_test_runner(self, cmd, binary):
```
Comment (mrobinson) [inline python/servo/testing_commands.py:472]: Please add typing information here.

File: python/servo/testing_commands.py [hunk 7]
```diff
@@ -402,16 +402,16 @@ def test_jquery(self, build_type: BuildType):
 
     @Command('test-dromaeo', description='Run the Dromaeo test suite', category='testing')
     @CommandArgument('tests', default=["recommended"], nargs="...", help="Specific tests to run")
-    @CommandBase.common_command_arguments(build_configuration=False, build_type=True)
-    def test_dromaeo(self, tests, build_type: BuildType):
-        return self.dromaeo_test_runner(tests, build_type)
+    @CommandBase.common_command_arguments(binary=True)
+    def test_dromaeo(self, tests, servo_binary):
```
Comment (mrobinson) [inline python/servo/testing_commands.py:406]:
```suggestion
    def test_dromaeo(self, tests, servo_binary: str):
```

File: python/servo/testing_commands.py [hunk 8]
```diff
@@ -402,16 +402,16 @@ def test_jquery(self, build_type: BuildType):
...
+        return self.dromaeo_test_runner(tests, servo_binary)
 
     @Command('update-jquery',
              description='Update the jQuery test suite expected results',
              category='testing')
-    @CommandBase.common_command_arguments(build_configuration=False, build_type=True)
-    def update_jquery(self, build_type: BuildType):
-        return self.jquery_test_runner("update", build_type)
+    @CommandBase.common_command_arguments(binary=True)
+    def update_jquery(self, servo_binary):
```
Comment (mrobinson) [inline python/servo/testing_commands.py:413]:
```suggestion
    def update_jquery(self, servo_binary: str):
```

File: python/servo/testing_commands.py [hunk 9]
```diff
@@ -319,13 +320,12 @@ def test_wpt_android(self, build_type: BuildType, binary_args=None, **kwargs):
             binary_args=self.in_android_emulator(build_type) + (binary_args or []),
             binary=sys.executable,
         )
-        return self._test_wpt(build_type=build_type, android=True, **kwargs)
+        return self._test_wpt(sys.executable, android=True, **kwargs)
 
     @CommandBase.allow_target_configuration
-    def _test_wpt(self, build_type: BuildType, with_asan=False, **kwargs):
+    def _test_wpt(self, servo_binary, **kwargs):
```
Comment (mrobinson) [inline python/servo/testing_commands.py:326]:
```suggestion
    def _test_wpt(self, servo_binary: str, **kwargs):
```

File: python/servo/testing_commands.py [hunk 10]
```diff
@@ -303,9 +304,9 @@ def test_wpt_failure(self, build_type: BuildType, **kwargs):
              description='Run the regular web platform test suite',
              category='testing',
              parser=wpt.create_parser)
-    @CommandBase.common_command_arguments(build_configuration=False, build_type=True)
-    def test_wpt(self, build_type: BuildType, with_asan=False, **kwargs):
-        return self._test_wpt(build_type=build_type, with_asan=with_asan, **kwargs)
+    @CommandBase.common_command_arguments(binary=True)
+    def test_wpt(self, servo_binary, **kwargs):
```
Comment (mrobinson) [inline python/servo/testing_commands.py:308]:
```suggestion
    def test_wpt(self, servo_binary: str, **kwargs):
```

File: python/servo/post_build_commands.py [hunk 11]
```diff
@@ -200,20 +198,15 @@ def android_emulator(self, args=None):
...
-                     help='Launch with specific binary')
-    @CommandArgument('--nightly', '-n', default=None,
-                     help='Specify a YYYY-MM-DD nightly build to run')
     @CommandArgument(
         'params', nargs='...',
         help="Command-line arguments to be passed through to Servo")
-    @CommandBase.common_command_arguments(build_configuration=False, build_type=True)
-    def rr_record(self, build_type: BuildType, bin=None, nightly=None, with_asan=False, params=[]):
+    @CommandBase.common_command_arguments(binary=True)
+    def rr_record(self, servo_binary, with_asan=False, params=[]):
```
Comment (mrobinson) [inline python/servo/post_build_commands.py:205]:
```suggestion
    def rr_record(self, servo_binary: str, with_asan=False, params=[]):
```

File: python/servo/post_build_commands.py [hunk 12]
```diff
@@ -74,17 +73,18 @@ class PostBuildCommands(CommandBase):
...
+    @CommandBase.common_command_arguments(build_configuration=False, build_type=False, binary=True)
     @CommandBase.allow_target_configuration
-    def run(self, params, build_type: BuildType, debugger=False, debugger_cmd=None,
-            headless=False, software=False, bin=None, emulator=False, usb=False, nightly=None, with_asan=False):
+    def run(self, servo_binary, params, debugger=False, debugger_cmd=None,
+            headless=False, software=False, emulator=False, usb=False):
+        self._run(servo_binary, params, debugger, debugger_cmd,
+                  headless, software, emulator, usb)
+
+    def _run(self, servo_binary, params, debugger=False, debugger_cmd=None,
```
Comment (mrobinson) [inline python/servo/post_build_commands.py:86]:
```suggestion
    def _run(self, servo_binary: str, params, debugger=False, debugger_cmd=None,
```

**Review by mrobinson — CHANGES_REQUESTED** [review]
Looks good. Just a couple minor comments:

File: python/servo/command_base.py [hunk 13]
```diff
@@ -591,9 +591,18 @@ def common_command_arguments(build_configuration=False, build_type=False):
                 )
             ]
 
+        if binary_selection:
+            decorators += [
+                CommandArgumentGroup('Binary select'),
```
Comment (mrobinson) [inline python/servo/command_base.py:596]:
```suggestion
                CommandArgumentGroup('Binary selection'),
```

**Review by mrobinson — APPROVED** [review]
Thanks. This is a nice cleanup.

**Pull request merged**